x-pack/legacy/plugins/infra/index.ts

```diff
@@ -67,8 +67,15 @@ export function infra(kibana: any) {
     config(Joi: typeof JoiNamespace) {
       return getConfigSchema(Joi);
     },
-    init(server: KbnServer) {
+    init(server: any) {
```

Comment:
This follows the same convention used in other plugins that call `server.addAppLinksToSampleDataset`

Reply:
So according to `LegacyPluginOptions` that function should have the signature of `InitPluginFunction` and `server` should be of type `Server`. APM's `index.ts` file seems to be close to the "right way". We can either fix this now by expanding the scope of your PR or fix it in a follow up PR. Either way, I think we should fix it so the typing works as expected. 

See: https://github.com/elastic/kibana/blob/master/src/legacy/plugin_discovery/types.ts#L87